platform/mellanox/get_component_versions/get_component_versions.py

```diff
@@ -91,35 +102,55 @@ def get_platform_component_versions():
 
 def get_current_version(comp):
     version = ""
-    # If there's only one command
-    if len(COMMANDS_FOR_ACTUAL[comp]) == 2:
-        version = subprocess.run(COMMANDS_FOR_ACTUAL[comp][0], shell=False, stdout=subprocess.PIPE, text=True)
-        version = str(version.stdout)
-    #If there are two commands and we need a pipe
-    elif len(COMMANDS_FOR_ACTUAL[comp]) == 3:
-        version = check_output_pipe(COMMANDS_FOR_ACTUAL[comp][0], COMMANDS_FOR_ACTUAL[comp][1])
-    parsed_version = re.search(COMMANDS_FOR_ACTUAL[comp][-1], version)
-    return parsed_version.group(1) if parsed_version else "N/A"
+    try:
+        # If there's only one command
+        if len(COMMANDS_FOR_ACTUAL[comp]) == 2:
+            version = subprocess.run(COMMANDS_FOR_ACTUAL[comp][0], shell=False, stdout=subprocess.PIPE, text=True)
+            version = str(version.stdout)
+        #If there are two commands and we need a pipe
+        elif len(COMMANDS_FOR_ACTUAL[comp]) == 3:
+            version = check_output_pipe(COMMANDS_FOR_ACTUAL[comp][0], COMMANDS_FOR_ACTUAL[comp][1])
+        parsed_version = re.search(COMMANDS_FOR_ACTUAL[comp][-1], version)
+        return parsed_version.group(1) if parsed_version else "N/A"
+    except Exception:
+        return "N/A"
 
 
 def format_output_table(table):
     return tabulate(table, HEADERS)
 
 
+def get_simx_version():
+    "SIMX": [["lspci", "-vv"], ["grep", "SimX"], "([0-9]+\\.[0-9]+-[0-9]+)"]
+    version = check_output_pipe(["lspci", "-vv"], ["grep", "SimX"])
+    parsed_version = re.search("([0-9]+\\.[0-9]+-[0-9]+)", version)
+    return parsed_version.group(1) if parsed_version else "N/A"
+
+
 def main():
 
     if os.getuid() != 0:
         print("Error: Root privileges are required")
         return
 
     compiled_versions = parse_compiled_components_file()
-    platform_versions = get_platform_component_versions()
+    simx_compiled_ver = compiled_versions.pop("SIMX")
 
+    # Add compiled versions to table
     output_table = []
     for comp in compiled_versions.keys():
         actual = get_current_version(comp)
         output_table.append([comp, compiled_versions[comp], actual])
 
+    # Handle if SIMX
+    simx_actual_ver = get_simx_version()
```

Comment:
@skr31 do we have a standard way to check whether it is a `SimX` platform? Platform API?

Reply:
I asked, this is the way someone from the simx team told to check

Reply:
There's a function in platform API, but we need to find the version anyway. Should I change it to use the function from device_data.py?

Reply:
@skr31 it is ok that you will have a separate call to get a version...but the platform check should be done from within the platform API code. BTW, maybe we can also use our platform API to get a SimX version? This code (get SimX version) looks generic to me and eventually it can be reused somewhere else

Reply:
@skr31 correct. Should be fine